Orange/widgets/data/oraclesql.py

```diff
@@ -0,0 +1,122 @@
+from AnyQt.QtWidgets import QSizePolicy, QPlainTextEdit, QHBoxLayout, QLineEdit
+from Orange.widgets import gui, widget, settings
+from Orange.widgets.widget import OWWidget
+from Orange.data import Table, Domain, ContinuousVariable, DiscreteVariable
+import pandas
```

Comment:
Pandas is not (as of yet) a requirement of the project. Make this import optional if pandas is available:
```py
try:
    from pandas import DataFrame
except ImportError:
    DataFrame = None
```

And for widget outputs:
```py
class OWOracleSQL(OWWidget):
    ...
    outputs = [('Data', Table)]
    if DataFrame is not None:
        outputs.append(('DataFrame', DataFrame))
```

And in `commit()`:
```py
if DataFrame is not None:
    df = DataFrame(...)
    self.send('DataFrame', df)
```

Reply:
You might want to migrate `outputs` signals to new-style, class-based signals: https://orange-development.readthedocs.io/widget.html#input-output-signal-definitions

Reply:
Here is a problem - i create orange table from dataframe (found the code somewhere). The widget does not work without pandas.

Reply:
There is a problem. The Table output is created from DataFrame. First you get a list of tuples, then convert them to DataFrame and then to Table.  If Pandas is not supported, the widget must be rebuilt.